keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,426 @@
+     * @throws com.azure.common.exception.ServiceRequestException when a secret with {@code secretBase.name} and {@code secretBase.version} doesn't exist in the key vault.
+     */
+    public Response<Secret> getSecret(SecretBase secretBase) {
+        Objects.requireNonNull(secretBase, "The Secret Base parameter cannot be null.");
+        return secretBase.version() == null ? service.getSecret(vaultEndpoint, secretBase.name(), "", API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block()
+            : service.getSecret(vaultEndpoint, secretBase.name(), secretBase.version(), API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Get the latest version of the specified secret from the key vault. The get operation is applicable to any secret stored in Azure Key Vault.
+     * This operation requires the {@code secrets/get} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * Secret secret = secretClient.getSecret("secretName").value();
+     * System.out.println(String.format("Secret is returned with name %s and value %s",secret.name(), secret.value()));
+     * </pre>
+     *
+     * @param name The name of the secret.
+     * @return A {@link Response} whose {@link Response#value()} contains the requested {@link Secret}.
+     * @throws com.azure.common.exception.ServiceRequestException when a secret with {@code name} doesn't exist in the key vault.
+     */
+    public Response<Secret> getSecret(String name) {
+        return getSecret(name, "");
+    }
+
+    /**
+     * Updates the attributes associated with the specified secret, but not the value of the specified secret in the key vault. The update
+     * operation changes specified attributes of an existing stored secret and attributes that are not specified in the request are left unchanged.
+     * The value of a secret itself cannot be changed. This operation requires the {@code secrets/set} permission.
+     *
+     * <p>The {@code secretBase} is required and its fields secretBase.name and secretBase.version cannot be null.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <pre>
+     * Secret secret = secretClient.getSecret("secretName").value();
+     * secret.notBefore(LocalDateTime.of(2030,01,01,12,00));
+     * SecretBase updatedSecretBase = secretClient.updateSecret(secret).value();
+     * Secret updatedSecret = secretClient.getSecret(updatedSecretBase.name()).value();
+     * </pre>
+     *
+     * @param secretBase the {@link SecretBase} object with updated properties.
+     * @throws NullPointerException if {@code secretBase} is {@code null}.
+     * @return A {@link Response} whose {@link Response#value()} contains the updated {@link SecretBase}.
+     * @throws com.azure.common.exception.ServiceRequestException when a secret with secretBase.name and secretBase.version doesn't exist in the key vault.
+     */
+    public Response<SecretBase> updateSecret(SecretBase secretBase) {
```

Comment:
I tend to prefer throughout the code that the variable name be `secret`, regardless of its type.

Reply:
updated,
But getSecret(SecretBase baseSecret) probably should not be changed. will be confusing to read its java docs as to we are passing in a secret to get a secret back.